### PR TITLE
added precision argument to ceil, floor and round

### LINKED ISSUE #393: Allow specifying precision with Math.{floor,round,ceil}
Add optional `digits` argument to `Math.floor`, `Math.round`, and `Math.ceil` to allow specifying rounding precision.

(This issue was tracked as PROD-7285 internally.)

### PR DESCRIPTION
**Pull request by rlgomes**
fixes #393

take primarily from an existing patch from the older jut-io/product
codebase and slightly adapted

### CODE REVIEW

**Comment by rlgomes** [comment]
@dmajda @bkutil

**Comment by dmajda** [comment]
+1

**Comment by dmajda** [comment]
@rlgomes FYI, I changed markup of level 1 and 2 headings in JuttleSpec tests to be `#`-based in #462. It would be great to adjust tests added here for consistency.

**Comment by rlgomes** [comment]
@dmajda I'll rebase with master and fix that up before merging.

**Comment by rlgomes** [comment]
Almost forgot to update the documentation... doing that now.

**Comment by rlgomes** [comment]
https://github.com/juttle/juttle/pull/455/files#diff-7ab98f8bedcc3d8b11f5e60f41664038 someone review the docs (@dmehra, @dmajda)

File: docs/modules/math.md [hunk 1]
```diff
@@ -113,9 +113,13 @@ Parameter  | Description   |  Required?
...
 
 ``` 
-Math.ceil(number)
+Math.ceil(number, digits)
 ```
 
+Parameter  | Description   |  Required?
+---------- | ------------- | ---------:
+`digits`   | A number indicating the number of significant digits to be used. When `digits` is 0, the number is rounded to the nearest integer;  `digits` greater than 0, then number is rounded to the specified number of decimal places; `digits` less than  0, the number is rounded to the left of the decimal point. | No
+
```
Comment (dmehra) [inline docs/modules/math.md]: Let's reword this a bit, how about:

The number of significant digits to be used. If 0, the value is rounded to the nearest integer; if greater than 0, the value is rounded to the specified number of decimal places; if less than 0, the value is rounded to the left of the decimal point.

**Comment by rlgomes** [comment]
merging on green

**Pull request merged**